### PR TITLE
Fix/statement logs

### PR DESCRIPTION
**Pull request by jean-humann**

### CODE REVIEW

**Comment by Minutis** [comment]
Hi,

thank your for initiative to contribute to this project. Could you please provide some information regarding the change you made?

File: .github/workflows/clean.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    types: [ closed ]
+
+jobs:
+  clean-registry:
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (pdambrauskas) [inline .github/workflows/clean.yml:14]: it would be nice to extract matrix to a single file ind include it on all workflows, that require it, if its possible.

File: .github/workflows/clean.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            hadoop_version: 3
+          - spark_version: 3.5.0
+            hadoop_version: 3
+    steps:
+      - name: Base ref
+        run: |
+            LOWER_BRANCH=${{ github.event.pull_request.head.ref }}
+            SLUG_BRANCH=$(echo "$LOWER_BRANCH" | sed -e 's/[^a-zA-Z0-9]/-/g' -e 's/--/-/g')
+            echo "BASE_REF=$SLUG_BRANCH" >> $GITHUB_ENV
+          fi
```
Comment (pdambrauskas) [inline .github/workflows/clean.yml:28]: there is no `if`, is this `fi` a leftover from the previous version?

File: .github/workflows/clean.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Delete tag on PR close or merge
+        if: github.event_name == 'pull_request' && github.event.action == 'closed'
+        env:
+          TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          IMAGE_NAME: ${{ github.repository }}
+          BASE_REF: ${{ env.BASE_REF }}
+        run: |
+          TAG_TO_DELETE=${IMAGE_NAME}:${BASE_REF}-spark${{ matrix.spark_version }}
+          TOKEN=$(echo "${TOKEN}" | awk '{print substr($0, 12)}')
```
Comment (pdambrauskas) [inline .github/workflows/clean.yml:37]: Why is this needed?

**Comment by pdambrauskas** [comment]
Pipeline is failing: 

> #30 ERROR: failed to push ghcr.io/exacaster/lighter:fix-statement-logs-spark3.3.3: unexpected status from POST request to https://ghcr.io/v2/exacaster/lighter/blobs/uploads/: 403 Forbidden

Something is wrong with the authorization

**Comment by pdambrauskas** [comment]
I've cherry-picked commits related to the fix. Regarding CI changes, we've chosen a different approach for selectively building images from Pull requests.

Thank you for your contribution!
